app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -562,6 +563,71 @@ def vm_compute_attributes(vm)
       vm_attrs
     end
 
+    def normalize_vm_attrs(vm_attrs)
+      normalized = slice_vm_attributes(vm_attrs, ['cpus', 'firmware', 'guest_id', 'annotation', 'resource_pool_id', 'image_id'])
+
+      normalized['cores_per_socket'] = vm_attrs['corespersocket']
+      normalized['memory'] = vm_attrs['memory_mb'].nil? ? nil : (vm_attrs['memory_mb'].to_i * 1024)
+
+      normalized['folder_path'] = vm_attrs['path']
+      normalized['folder_name'] = self.folders.detect { |f| f.path == normalized['folder_path'] }.try(:name)
+
+      normalized['cluster_id'] = self.available_clusters.detect { |c| c.name == vm_attrs['cluster'] }.try(:id)
+      normalized['cluster_name'] = vm_attrs['cluster']
+      normalized['cluster_name'] = nil if normalized['cluster_name'].empty?
+
+      if normalized['cluster_name']
+        normalized['resource_pool_id'] = self.resource_pools(:cluster_id => normalized['cluster_name']).detect { |p| p.name == vm_attrs['resource_pool'] }.try(:id)
+      end
+      normalized['resource_pool_name'] = vm_attrs['resource_pool']
+      normalized['resource_pool_name'] = nil if normalized['resource_pool_name'].empty?
+
+      normalized['guest_name'] = self.guest_types[vm_attrs['guest_id']]
+
+      normalized['hardware_version_id'] = vm_attrs['hardware_version']
+      normalized['hardware_version_name'] = vm_hw_versions[vm_attrs['hardware_version']]
+
+      normalized['memory_hot_add_enabled'] = to_bool(vm_attrs['memoryHotAddEnabled'])
+      normalized['cpu_hot_add_enabled'] = to_bool(vm_attrs['cpuHotAddEnabled'])
+      normalized['add_cdrom'] = to_bool(vm_attrs['add_cdrom'])
```

Comment:
This is an arbitrary parameter we introduced and should only be set during vm creation. I would omit it here.

Reply:
Does it mean it can't be saved into attributes of a compute profile? AFAIK it's available in the compute profile UI. If we're removing it from here, we should remove it from the profile UI as well.

Reply:
It can be saved into a compute profile, no worries here. I just wanted to say, that it's not an attribute that can be read from VM data.